plutus-core/plutus-core/src/PlutusCore/DeBruijn/Internal.hs

```diff
@@ -119,11 +119,10 @@ newtype TyDeBruijn = TyDeBruijn DeBruijn
 instance Wrapped TyDeBruijn
 
 instance HasPrettyConfigName config => PrettyBy config NamedDeBruijn where
-    prettyBy config (NamedDeBruijn txt0 (Index ix))
-        | showsUnique = pretty txt <> "_i" <> pretty ix
-        | otherwise   = pretty txt
+    prettyBy config (NamedDeBruijn txt (Index ix))
+        | showsUnique = pretty . toPrintedName $ txt <> "_i" <> render (pretty ix)
```

Comment:
Given how weird this is, I think it's worth recording our discussion in a note so the reasons for doing this weird thing are recorded.